drivers/kscan/kscan_mchp_xec.c

```diff
@@ -57,31 +70,84 @@ struct kscan_xec_data {
 	K_KERNEL_STACK_MEMBER(thread_stack, TASK_STACK_SIZE);
 };
 
-static KSCAN_Type *base = (KSCAN_Type *)
-	(DT_INST_REG_ADDR(0));
+#ifdef CONFIG_SOC_SERIES_MEC172X
+static void kscan_clear_girq_status(const struct device *dev)
+{
+	struct kscan_xec_config const *cfg = dev->config;
+
+	mchp_xec_ecia_girq_src_clr(cfg->girq, cfg->girq_pos);
+}
+
+static void kscan_girq_ctrl(const struct device *dev, int enable)
```

Comment:
Non blocking suggestion: name this `void kscan_configure_girq(const struct device *dev, bool enable)`, or dropping the `enable` parameter and creating `kscan_enable_girq()` and `kscan_disable_girq()`.

Reply:
Changed to `void kscan_configure_girq(const struct device *dev, bool enable)` as per suggestion